src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+
+import { SavedObjectsErrorHelpers } from '../../../../core/server';
+
+import { CryptographyClient } from '../cryptography';
+
+import { Credential } from '../../common';
+
+/**
+ * Describes the Credential Saved Objects Client Wrapper class,
+ * which contains the factory used to create Saved Objects Client Wrapper instances
+ */
+export class CredentialSavedObjectsClientWrapper {
+  private type: string = 'credential';
+  private authType: Credential.SharedAuthType = 'shared';
```

Comment:
what is `shared` auth type?

Reply:
As previously discussed, credentials can have different authTypes (Shared / Private / etc.). We will only support shared for current release and will explicitly notify customers that this credential is shared for everyone.

Reply:
would `Credential.AuthType = 'shared | private | etc'` make more sense and be more long-term sustainable here? (as opposed to SharedAuthType, which feels a bit redundant here)

Reply:
yeah, there's some syntax level optimize we can do. But I am thinking if we shall put authType in data source instead of credential

Reply:
> would `Credential.AuthType = 'shared | private | etc'` make more sense and be more long-term sustainable here? (as opposed to SharedAuthType, which feels a bit redundant)

use enum instead